finetune/base_models/long_doc/encoder.py

```diff
@@ -0,0 +1,77 @@
+import os
+import json
+import regex as re
+import logging
+import time
+from functools import lru_cache
+
+import numpy as np
+
+import finetune
+import spacy
+from finetune.encoding.input_encoder import BaseEncoder, EncodedOutput, get_pairs
+
+import unicodedata
+
+FINETUNE_FOLDER = os.path.dirname(finetune.__file__)
+LOGGER = logging.getLogger("finetune")
+
+class LongDocEncoder(BaseEncoder):
+    UNK_IDX = 0
+
+    def __init__(self, encoder_path=None, vocab_path=None):
+        self.nlp = None
+        self.initialized = False
+        pass
+
+    def _lazy_init(self):
+        if not self.initialized:
+            self.nlp = spacy.load("en_vectors_web_lg")
+            self.start_token = np.zeros([300], np.float32)
+            self.delimiter_token = self.start_token
+            self.mask_token = self.start_token
+            self.end_token = self.start_token
+        self.initialized = True
+
+    def _encode(self, texts, stochastic=False):
```

Comment:
Because this sourcemodel takes in different data (i.e. here texts is a 2-tuple), we should probably have some type check at a reasonable point, just so the user doesn't hit some confusing error somewhere else?